desc/objectives/objective_funs.py

```diff
@@ -136,17 +90,13 @@ def jit(self):  # noqa: C901
             "vjp_scaled_error",
             "vjp_unscaled",
         ]
-
         for method in methods:
             try:
-                delattr(self, method)
+                setattr(
+                    self, method, functools.partial(getattr(self, method)._fun, self)
```

Comment:
I was going to ask why grad jac etc had the `Derivative` call inside of them now, but this functools.partial call basically partially evaluates the method with`self`, which also creates the Derivative and sets up the correct AD method, so that subsequent calls are not re-creating the grad/jac function again, but rather just using it?

Reply:
This code is only for **un**jitting something (ie if `use_jit=False`) in which case it will effectively create a new `Derivative` object on each call (`partial` doesn't actually evaluate anything when its created).

The default (when jitted) is that the creation of the `Derivative` object basically gets compiled away by jax so its not costing anything. And even if it's created each time instantiating the object is negligible compared to actually computing the derivative.